### DISCUSSION TITLE
In .dart , how to write A & (B | C) in where

### DISCUSSION THREAD

**Discussion by hcanyz** [simolus3/drift#2381] [Q&A]
Excuse me, this may be a naive question, but I don't know the answer :(

I need to write a condition of a Where statement in .dart like:

```dart
where ((t) {
 final column1Exp = t.column1.equals(canvasId);
 final column2Exp =
     t.column2.isSmallerThanValue(10) & t.column2.isBiggerThanValue(0);
 final column3Exp =
     t.column3.isSmallerThanValue(10) & t.column3.isBiggerThanValue(0);
 return column1Exp & (column2Exp | column3Exp);
})
```

The sql it finally executes is like this

```sql
SELECT * FROM "table1" WHERE "column1" = ? AND ("column2" < ? AND "column2" > ? OR "column3" < ? AND "column3" > ?);
```

This is not as expected, which I would expect

```sql
SELECT * FROM "table1" WHERE "column1" = ? AND (("column2" < ? AND "column2" > ?) OR ("column3" < ? AND "column3" > ?));
```

I found out that ```BaseInfixOperator``` might solve this problem, but it's internal.

**Comment by hcanyz** [comment]
I was mistaken, they are equivalent = = .

repeated with this: https://github.com/simolus3/drift/discussions/1544

**Reply by simolus3** [reply]
You made me double-check this too :) But yes, `OR` has a lower precedence which is why the parentheses are unnecessary.

By the way, if the columns are integers your expression can also be written as

```dart
where ((t) {
 final column1Exp = t.column1.equals(canvasId);
 final column2Exp = t.column2.isBetweenValues(1, 9);
 final column3Exp = t.column3.isBetweenValues(1, 9);
 return column1Exp & (column2Exp | column3Exp);
})
```

Maybe that makes both the Dart source and the generated SQL easier to read.
